pkg/otlp/model/translator/statspayload.go

```diff
@@ -0,0 +1,471 @@
+			DBType:         k.DBType,
+			Synthetics:     k.Synthetics,
+			Hits:           v.Hits,
+			Errors:         v.Errors,
+			Duration:       v.Duration,
+			OkSummary:      v.OkSummary,
+			ErrorSummary:   v.ErrorSummary,
+			TopLevelHits:   v.TopLevelHits,
+		})
+	}
+	return cgs
+}
+
+// UnsetHostnamePlaceholder is the string used as a hostname when the hostname can not be extracted from span attributes
+// by the processor. Upon decoding the metrics, the Translator will use its configured fallback SourceProvider to replace
+// it with the correct hostname.
+//
+// This isn't the most ideal approach to the problem, but provides the better user experience by avoiding the need to
+// duplicate the "exporter::datadog::hostname" configuration field as "processor::datadog::hostname". The hostname can
+// also not be left empty in case of failure to obtain it, because empty has special meaning. An empty hostname means
+// that we are in a Lambda environment. Thus, we must use a placeholder.
+const UnsetHostnamePlaceholder = "__unset__"
+
+// StatsPayloadToMetrics converts OTLP Metrics to an APM Stats Payload.
+func (t *Translator) StatsPayloadFromMetrics(mx pmetric.Metrics) (pb.StatsPayload, error) {
+	var stats pb.StatsPayload
+	rmxs := mx.ResourceMetrics()
+	for i := 0; i < rmxs.Len(); i++ {
+		rmx := rmxs.At(i)
+		attr := rmx.Resource().Attributes()
+		if v, ok := attr.Get(keyAPMStats); !ok || !v.Bool() {
+			t.logger.Debug(fmt.Sprintf("StatsPayload Translator: was asked to convert metrics to stats payload, but identifier key %q was not present. Skipping.", keyAPMStats))
+			continue
+		}
+		hostname := getStr(attr, statsKeyHostname)
+		tags := strings.Split(getStr(attr, statsKeyTags), ",")
+		if hostname == UnsetHostnamePlaceholder {
+			src, err := t.source(attr)
+			if err != nil {
+				return pb.StatsPayload{}, err
+			}
+			switch src.Kind {
+			case source.HostnameKind:
+				hostname = src.Identifier
+			case source.AWSECSFargateKind:
+				hostname = ""
+				tags = append(tags, src.Tag())
+			}
+		}
+		cp := pb.ClientStatsPayload{
+			Hostname:         hostname,
+			Env:              getStr(attr, statsKeyEnv),
+			Version:          getStr(attr, statsKeyVersion),
+			Lang:             getStr(attr, statsKeyLang),
+			TracerVersion:    getStr(attr, statsKeyTracerVersion),
+			RuntimeID:        getStr(attr, statsKeyRuntimeID),
+			Sequence:         getInt(attr, statsKeySequence),
+			AgentAggregation: getStr(attr, statsKeyAgentAggregation),
+			Service:          getStr(attr, statsKeyService),
+			ContainerID:      getStr(attr, statsKeyContainerID),
+			Tags:             tags,
+		}
+		smxs := rmx.ScopeMetrics()
+		for j := 0; j < smxs.Len(); j++ {
+			mxs := smxs.At(j).Metrics()
+			var (
+				buck pb.ClientStatsBucket
+				agg  aggregations
+			)
+			for k := 0; k < mxs.Len(); k++ {
+				m := mxs.At(k)
+				switch m.Type() {
+				case pmetric.MetricTypeSum:
+					key, val := t.extractSum(m.Sum(), &buck)
+					switch m.Name() {
+					case "hits":
```

Comment:
Do we choose these names? If so, I think we should namespace them (`dd.stats.hits`?) to avoid collisions